core/assignment.go

```diff
@@ -189,3 +227,8 @@ func roundUpDivide(a, b uint) uint {
 	return (a + b - 1) / b
```

Comment:
Does this need to handle the case where `a + b` is already overflow the uint?

Reply:
I think we should only be using uints here in cases where this won't overflow.